mempool/mempool.go

```diff
@@ -296,6 +298,10 @@ func (mem *Mempool) TxsWaitChan() <-chan struct{} {
 //     It gets called from another goroutine.
 // CONTRACT: Either cb will get called, or err returned.
 func (mem *Mempool) CheckTx(tx types.Tx, cb func(*abci.Response)) (err error) {
+	return mem.checkTxFromPeer(tx, cb, 0)
+}
+
+func (mem *Mempool) checkTxFromPeer(tx types.Tx, cb func(*abci.Response), peerID uint16) (err error) {
```

Comment:
Would it make sense to create a struct

```go
type txInfo struct {
  PeerID uint16
}
```

and use it instead of raw uint16?

Reply:
I'd prefer if we spec'd out some more details about what else would go here before making more abstraction. Its not clear to me why we would want time to always live in the cache for example.